charts/operator-wandb/charts/weave-trace/templates/migrate-hook.yaml

```diff
@@ -0,0 +1,68 @@
+{{ if .Values.migration.useHook }}
```

Comment:
We may be able to remove this.

The problem with this approach is that it doesn't work as a pre-install hook because it depends on the `Secret` containing the clickhouse credentials, and those don't yet exist when the job container starts.  It works fine if the `Secret` is provisioned ahead of time.  The benefit to this approach is it is technically more correct since we only want a single migration container running.

The alternative, the `initContainer` associated with the `weave-trace` pod, runs an init container for every pod.  This is mostly OK since the migration script quickly NOOPs when there is no need to migrate.

Reply:
Can you also add the creation of the secret to the hook at the root level? Likewise we could add the creation of those secrets in this sub chart